qfi_opt/spin_models.py

```diff
@@ -374,28 +372,40 @@ def act_on_subsystem(num_qubits: int, op: np.ndarray, *qubits: int) -> np.ndarra
 def get_jacobian_func(
     simulate_func: Callable,
     *,
-    disable_jax: bool = DISABLE_JAX,
-    step_sizes: float | Sequence[float] = 1e-4,
+    disable_diffrax: bool = DISABLE_DIFFRAX,
+    step_sizes: float | Sequence[float] = 1e-6,
 ) -> Callable:
     """Convert a simulation method into a function that returns its Jacobian."""
 
-    if not DISABLE_JAX:
-        jacobian_func = jax.jacrev(simulate_func, argnums=(0,), holomorphic=True)
-
-        def get_jacobian(*args: object, **kwargs: object) -> np.ndarray:
-            return jacobian_func(*args, **kwargs)[0]
+    if not DISABLE_DIFFRAX:
+
+        def get_jacobian(params: Sequence[float], *args: object, **kwargs: object) -> np.ndarray:
+            primals, vjp_func = jax.vjp(simulate_func, params, *args)
+            result = np.zeros((primals.shape[1], primals.shape[0], len(params)), dtype=COMPLEX_TYPE)
+            for i in range(primals.shape[1]):
+                for j in range(primals.shape[0]):
```

Comment:
```suggestion
            for j, i in np.ndindex(primals.shape):
```
(and also maybe `i` and `j` should be switched?)

Reply:
I went ahead and made this change